lib/active_recall/algorithms/sm2.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module ActiveRecall
+  class SM2
+    MIN_EASINESS_FACTOR = 1.3
+
+    def self.score(box:, easiness_factor:, times_right:, times_wrong:, grade:, current_time: Time.current)
+      new(
+        box: box,
+        easiness_factor: easiness_factor,
+        times_right: times_right,
+        times_wrong: times_wrong,
+        grade: grade,
+        current_time: current_time
+      ).score
+    end
+
+    def initialize(box:, easiness_factor:, times_right:, times_wrong:, grade:, current_time: Time.current)
+      @box = box
+      @easiness_factor = easiness_factor || 2.5
+      @times_right = times_right
+      @times_wrong = times_wrong
+      @grade = grade
```

Comment:
We should raise if we get a grade outside of possible bounds.

Reply:
Fixed [here](https://github.com/jaysonvirissimo/active_recall/pull/25/commits/d787541b93ed38ce46a0d74039f07326e6224ac8).